lib/puma/plugin/statsd.rb

```diff
@@ -86,25 +27,46 @@ def max_threads
   # We can start doing something when we have a launcher:
   def start(launcher)
     @launcher = launcher
-
-    @statsd = ::StatsdConnector.new
-    if @statsd.enabled?
-      @launcher.events.debug "statsd: enabled (host: #{@statsd.host})"
-
-      # Fetch global metric prefix from env variable
-      @metric_prefix = ENV.fetch("STATSD_METRIC_PREFIX", nil)
-      if @metric_prefix && !@metric_prefix.end_with?(::StatsdConnector::METRIC_DELIMETER)
-        @metric_prefix += ::StatsdConnector::METRIC_DELIMETER
-      end
+    host = ENV.fetch("STATSD_HOST", nil)
+    port = ENV.fetch("STATSD_PORT", 8125)
+    
+    if host
+      @statsd_client = Datadog::Statsd.new(host, port, tags: environment_variable_tags)
+      @launcher.events.debug "statsd: enabled (host: #{host})"
 
       register_hooks
     else
-      @launcher.events.debug "statsd: not enabled (no #{StatsdConnector::ENV_NAME} env var found)"
+      @launcher.events.debug "statsd: not enabled (no STATSD_HOST env var found)"
     end
   end
 
   private
 
+  # Send data to statsd every few seconds
+  def stats_loop
+    sleep 5
+    loop do
+      @launcher.events.debug "statsd: notify statsd"
+      begin
+        stats = ::PumaStats.new(fetch_stats)
+
+        @statsd_client.batch do |s|
+          s.gauge('puma.workers', stats.count_value_for_key(:workers, 1))
+          s.gauge('puma.booted_workers', stats.count_value_for_key(:booted_workers, 1))
+          s.gauge('puma.running', stats.count_value_for_key(:running))
+          s.gauge('puma.backlog', stats.count_value_for_key(:backlog))
+          s.gauge('puma.pool_capacity', stats.count_value_for_key(:pool_capacity))
+          s.gauge('puma.max_threads', stats.count_value_for_key(:max_threads))
+          s.gauge('puma.requests_count', stats.count_value_for_key(:requests_count))
```

Comment:
does this fail gracefully in case this is used with puma < 5?

Reply:
it'll return 0 in this case.